src/main/java/sdu/sem2/se17/persistence/data/ProductionCompanyHandlerImpl.java

```diff
@@ -18,26 +22,118 @@ public ProductionCompanyHandlerImpl(DataSource dataSource) {
 
     @Override
     public Optional<ProductionCompany> create(ProductionCompany productionCompany) {
-        return Optional.empty();
+        Optional<ProductionCompany> result = Optional.empty();
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement statement = connection.prepareStatement(""" 
+                            INSERT INTO ProductionCompany (name)
+                            VALUES (?)
+                            RETURNING *
+                        """)
+        ) {
+            configureStatement(productionCompany, statement);
+
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()) {
+                    result = Optional.of(map(resultSet));
+                }
+            }
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+
+        return result;
     }
 
     @Override
     public Optional<ProductionCompany> read(long id) {
-        return Optional.empty();
+        Optional<ProductionCompany> result = Optional.empty();
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement statement = connection.prepareStatement(""" 
+                            SELECT * FROM ProductionCompany
+                            WHERE id = ?
+                            RETURNING *
+                        """)
+        ) {
+            statement.setLong(1, id);
+
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()) {
+                    result = Optional.of(map(resultSet));
+                }
+            }
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+
+        return result;
     }
 
     @Override
     public void update(ProductionCompany productionCompany) {
-
+        try (
+                Connection connection = dataSource.getConnection();
+                PreparedStatement statement = connection.prepareStatement(""" 
+                            UPDATE ProductionCompany
+                            SET (name) = name
```

Comment:
Remove second "name"
```suggestion
                            SET (name) =
```